src/util.c

```diff
@@ -0,0 +1,22 @@
+#include "util.h"
+#include <stdint.h>
+
+bool is_big_endian(void) {
+  static const uint32_t endianess = 1;
+  return (*(uint8_t *)&endianess);
+}
```

Comment:
I'm looking at this and thinking it has the logic backwards. On our little-endian ARM microcontrollers, `endianess` (which should be spelled endianness, with 2 Ns) will be represented with the lowest byte in memory as 1, and the next 3 increasing memory addresses as 0. So `&endianess` will refer to that lowest memory address, which, when cast to a `uint8_t *` will be 1. Right? Should be easy to add a unit test.

So, either we could keep the implementation and rename the function `is_little_endian`, or we could keep the name and do some pointer math, adding 3 after the cast to `uint8_t *`. I think I'd prefer to keep the implementation and change the name since the additional pointer math makes me nervous if it's not necessary.

Reply:
Agreed, and correct. Was a mistake in implementation and thinking about numbers and backwards bits :)